text/0022-remotes.md

```diff
@@ -0,0 +1,98 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Some definitions used in this document:
+
+* Registry - the Qri-run service for pinning, searching, etc
+* Remote - the future functionality included in the main Qri binary
+* Client - a Qri command-line program that get commands from stdin
+* Backend - a Qri program that is not the Electron frontend
+
+Currently, the Registry exists to solve multiple problems centered around a distributed system's lack of centralization:
+
+* User identity and registration
+* Search
+* Availability
+
+The Registry acts in a "semi-centralized" way: it provides these types of features, but is not strictly required to use Qri by itself. However, there is currently no way for an average user to run something like the Registry on their own.
+
+Currently functionality that exists only in the Registry now includes:
+
+* `POST /dataset` to upload a dataset's head
+* `/search` to search over known datasets (online or not)
+* `/datasets` to list all known datasets (online or not)
+* `/profile` user information
+* `/reputation` user trustability
+* `/pins` pinning a dataset
+* `/dsync` data block uploading
+
+Long-term we want to obliviate the need for these to be in a separate executable.
+
+The plan of action is as follows:
+
+* Add a flag to Qri's backend to run it as a remote. This will work similarily to the current "read-only" flag, disabling most write functionality.
+* Add APIs to the backend that search a similar purpose to what the Registry is doing now: `POST /dataset` in particular.
+* Get enough APIs in the backend such that it can be run as a remote which provides availability to users that want it.
+* Add functionality to Qri's command-line to publish to this remote instead of, or in addition to the public Registry.
+* Over time, move the rest of the functionality from the Registry into the backend remote.
+* Replace the live Registry with a standard remote.
+* Delete old Registry code.
+* New semi-centralized features get added to the Remote feature set.
+
+Some specific notes as to how this is presented. First, the concept of the Registry is a good way to explain Qri to users: it exists as a way to keep data online even if your own node is not. This shouldn't change as we move to Remotes; instead of deleting the _concept_ of a Registry, we're generalizing it by introduced a Remote as something you can run yourself if you're an advanced user with a special use-case. Secondly, since a Remote is considered an expert-level concept, it's okay for its UX to be slightly more advanced than it may be with other features. The easy thing to use should remain to be the Qri run public Registry. This implies, for example, that setting up a client to use a remote should be more like `git config set remote <url>` instead of `git remote add <name> <url>`, with `remote` as a top-level command.
+
```

Comment:
I think we should add an explicit section here

### Access Control / Remote Rulesets
Currently, we don't have a robust system for declaring rules about what datasets a remote should accept. Defining these rules will require additional RFCs to map the problem space before dev work can begin.

Until robust ruleset declaration exists, remotes will operate as an opt-in experiment for Qri users, with very limited means of configuring what data to accept. Users operating a remote should be prepared to manually monitor the remote they're operating for data added there.